modules/apps/frontend-editor/frontend-editor-alloyeditor-web/src/main/resources/META-INF/resources/js/alloyeditor_source.js

```diff
@@ -87,17 +87,20 @@ AUI.add(
 				_getEditorStateLexiconIcon() {
 					var instance = this;
 
-					var currentState = MAP_TOGGLE_STATE[instance._isVisible];
+					var icon;
 
-					var icon = currentState.icon;
+					var currentState = MAP_TOGGLE_STATE[instance._isVisible];
 
-					if (!icon) {
+					if (!currentState.icon) {
 						icon = Liferay.Util.getLexiconIcon(
 							currentState.iconCssClass
 						);
 
 						currentState.icon = icon;
 					}
+					else {
+						icon = currentState.icon.cloneNode(true);
+					}
```

Comment:
Change looks good to me and the description makes sense.

One nit: might be more readable if you invert the order of the condition to avoid the negation, which is harder to read.

```js
if (currentState.icon) {
    icon = currentState.icon.cloneNode(true);
}
else {
    icon = Liferay.Util.getLexiconIcon(
        currentState.iconCssClass
    );

    currentState.icon = icon;
}
```

Reply:
Hey,
Sorry I somehow missed this PR, I just pushed the requested change.

Thanks for the review